osh/cmd_parse.py

```diff
@@ -1106,6 +1107,18 @@ def ParseSimpleCommand(self):
             return command.ShAssignment(left_tok, pairs, redirects)
 
         kind, kw_token = word_.IsControlFlow(suffix_words[0])
+
+        if (typed_args is not None and kw_token is not None and
+                kw_token.id == Id.ControlFlow_Return):
```

Comment:
These checks look good -- are they tested?

Maybe add some stuff to `test/ysh-runtime-error.sh`, or a new `test/ysh-parse-errors.sh` ?